### PR TITLE
Added auto-reviewer config file

### PR DESCRIPTION
**Pull request by gaaclarke**

### CODE REVIEW

File: .github/auto_assign.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Set to true to add reviewers to pull requests
```
Comment (cbracken) [inline .github/auto_assign.yml]: Maybe add a short comment at the top saying that this is a config file for the [insert name of github app here] bot whose purpose is to auto-assign a reviewer in order to better help distribute reviews across the team, and familiarise developers with parts of the engine they may not typically deal with.

File: .github/auto_assign.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+addReviewers: true
+
+# Set to true to add assignees to pull requests
+addAssignees: false
+
+# A list of reviewers to be added to pull requests (GitHub user name)
```
Comment (cbracken) [inline .github/auto_assign.yml]: Add a note that engine committers should be added here upon joining the team (and maybe a link to CONTRIBUTING.md for people who want to see what that entails)

Comment (gaaclarke) [reply .github/auto_assign.yml]: Done, will add contributing note when this lands and the bot is installed.

File: .github/auto_assign.yml (hunk 1 above)
Comment (gaaclarke) [reply .github/auto_assign.yml]: done

**Comment by chinmaygarde** [comment]
How is this different from a CODEOWNERS file? That seems to not require a bot https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners.

**Comment by gaaclarke** [comment]
> How is this different from a CODEOWNERS file? That seems to not require a bot https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners.

From my understanding from skimming CODEOWNERS and talking to Chris.  CODEOWNERS makes it so we gate checkins on someones approval.  Auto-assign grabs someone from a pool of candidates at random and assigns them to review PRs.

**Pull request merged**